core/network/src/debug_info.rs

```diff
@@ -76,7 +76,7 @@ impl<TSubstream> DebugInfoBehaviour<TSubstream> {
 			ping: Ping::new(PingConfig::new()),
 			identify,
 			nodes_info: FnvHashMap::default(),
-			garbage_collect: Box::new(Interval::new(GARBAGE_COLLECT_INTERVAL).map(|()| Ok(())).compat()),
+			garbage_collect: Box::new(interval(GARBAGE_COLLECT_INTERVAL).map(|()| Ok(())).compat()),
```

Comment:
That should work, I think (with better formatting):

```suggestion
			garbage_collect: Box::new(futures::stream::unfold((), |()| async { Delay::new(GARBAGE_COLLECT_INTERVAL).await; ((), ()) }).map(|()| Ok(())).compat()),
```